src/Codeception/Command/Shared/FileSystem.php

```diff
@@ -60,12 +60,26 @@ protected function removeSuffix($classname, $suffix)
         return preg_replace("~$suffix$~", '', $classname);
     }
 
-    protected function save($filename, $contents, $force = false, $flags = null)
+    protected function save($filename, $contents, $force = false)
     {
         if (file_exists($filename) && !$force) {
             return false;
         }
-        file_put_contents($filename, $contents, $flags);
+
+        //Write in a locked temporary file in order to reduce non thread-safe disk instructions,
+        //  and reduce risk of `Bus Errors` when running tests in parallel.
+        //@see https://github.com/Codeception/Codeception/issues/2054#event-1046859271
+        $tmpFilename = $filename . '.tmp';
```

Comment:
you're right about using a ramdom name, to avoid collision but `tempnam` create the file on the disk, which is not-needed (file_put_contents actually create the file).

I'm upgrading this code to use uniqid() in $tmpFilename